packages/repository/src/repositories/legacy-juggler-bridge.ts

```diff
@@ -262,6 +265,38 @@ export class DefaultCrudRepository<
     );
   }
 
+  protected createHasManyThroughRepositoryFactoryFor<
+    TargetEntiy extends Entity,
+    TargetID,
+    ThroughEntity extends Entity,
+    ThroughID,
+    ForeignKeyType
+  >(
+    relationName: string,
+    targetRepositoryGetter: Getter<EntityCrudRepository<TargetEntiy, TargetID>>,
+    throughRepositoryGetter: Getter<
+      EntityCrudRepository<ThroughEntity, ThroughID>
+    >,
+  ): HasManyThroughRepositoryFactory<
+    TargetEntiy,
+    TargetID,
+    ThroughEntity,
+    ForeignKeyType
```

Comment:
Not sure about `ForeignKeyType`. Do we really need `ThroughID` and `ForeignKeyType`?
@KalleV PTAL

Reply:
@derdeka I think the ForeignKeyType is needed. It is consistent with the HasManyRepositoryFactory typing. For example:

```
export class UserRepository<
  User,
  typeof User.prototype.id
> {
  public readonly groups: HasManyThroughRepositoryFactory<
    Group,
    UserGroup,
    typeof Group.prototype.id
  >;
}
```